node/src/components/deploy_acceptor.rs

```diff
@@ -101,120 +184,586 @@ impl DeployAcceptor {
+                        let (query_key, path) = match contract_identifier.clone() {
+                            ContractIdentifier::Name(name) => {
+                                let query_key: Key = account_hash.into();
+                                let path = vec![name];
+                                (query_key, path)
+                            }
+                            ContractIdentifier::Hash(contract_hash) => {
+                                let query_key: Key = contract_hash.into();
+                                (query_key, vec![])
+                            }
+                        };
+                        effect_builder
+                            .get_contract_for_validation(prestate_hash, query_key, path)
+                            .event(move |maybe_contract| Event::GetContractResult {
+                                event_metadata,
+                                prestate_hash,
+                                is_payment: true,
+                                contract_identifier,
+                                maybe_contract,
+                                verification_start_timestamp,
+                            })
+                    }
+                    ExecutableDeployItemIdentifier::Package(contract_package_identifier) => {
+                        let (query_key, path): (Key, Vec<String>) =
+                            match contract_package_identifier.clone() {
+                                ContractPackageIdentifier::Name { name, .. } => {
+                                    let query_key: Key = account_hash.into();
+                                    let path = vec![name];
+                                    (query_key, path)
+                                }
+                                ContractPackageIdentifier::Hash {
+                                    contract_package_hash,
+                                    ..
+                                } => {
+                                    let query_key = contract_package_hash.into();
+                                    (query_key, vec![])
+                                }
+                            };
+                        effect_builder
+                            .get_contract_package_for_validation(prestate_hash, query_key, path)
+                            .event(
+                                move |maybe_contract_package| Event::GetContractPackageResult {
+                                    event_metadata,
+                                    prestate_hash,
+                                    is_payment: true,
+                                    contract_package_identifier,
+                                    maybe_contract_package,
+                                    verification_start_timestamp,
+                                },
+                            )
+                    }
                 }
             }
+        }
+    }
+
+    fn verify_session_logic<REv: ReactorEventT>(
```

Comment:
I don't think we should be verifying the session logic at all. If the deploy can be paid for we shouldn't really care whether deploy's creator made sure it can also be executed.